addons/stock/models/stock_picking.py

```diff
@@ -1257,6 +1257,7 @@ def _create_backorder(self):
                 moves_to_backorder.move_line_ids.package_level_id.write({'picking_id': backorder_picking.id})
                 moves_to_backorder.mapped('move_line_ids').write({'picking_id': backorder_picking.id})
                 backorders |= backorder_picking
+                backorder_picking.user_id = False
```

Comment:
That was my initial idea, but in that case a duplicated stock picking would not have the same user as the original (and I believe it should). The requirement (as described in the other task) was to only limit this change to backorders, and it makes sense in my opinion.

If we're OK with not copying the user at all, I can implement it with `copy=False` :)

Reply:
OK, I checked with THD and it makes sense not to copy user ID at all (not only in backorders). I will update the code accordingly :)

Reply:
@amoyaux I've updated the code, according to the following requirements:
* created by Pickings -> New: should have current user assigned
* duplicated by Pickings -> existing one -> duplicate: no user assigned
* created by validating a PO/SO: no user assigned

It turned out to be a bit more complicated, because by default we were setting the `user_id` field to `self.env.user`, so even if it was not copied, it was still assigned a value. I've changed that to use `compute=` instead of `default=`, could you please check if this solution is OK?

Reply:
My previous solution broke a few tests and I realized it's not the correct one :( Instead of using `compute=` I decided to override the `copy()` method and make sure that we don't use a default `user_id` while copying. Is this OK @amoyaux ?